### PR TITLE
Move Android instrumented test `ThemeTest` to unit test

### PR DESCRIPTION
**Pull request by SimonMarquis**
This was the only remaining instrumented test of `:core:designsystem` module.

### CODE REVIEW

File: core/designsystem/src/test/kotlin/com/google/samples/apps/nowinandroid/core/designsystem/ThemeTest.kt [hunk 1]
```diff
@@ -53,6 +56,8 @@ import kotlin.test.assertEquals
  * [LocalBackgroundTheme] — have the expected values for a given theme mode, as specified by the
  * design system.
  */
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [33])
```
Comment (SimonMarquis) [inline core/designsystem/src/test/kotlin/com/google/samples/apps/nowinandroid/core/designsystem/ThemeTest.kt:60]: Waiting on #1047 to remove this line
```suggestion
```

File: core/designsystem/src/test/kotlin/com/google/samples/apps/nowinandroid/core/designsystem/ThemeTest.kt [hunk 2]
```diff
@@ -43,6 +43,9 @@ import com.google.samples.apps.nowinandroid.core.designsystem.theme.NiaTheme
 import com.google.samples.apps.nowinandroid.core.designsystem.theme.TintTheme
 import org.junit.Rule
 import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
```
Comment (SimonMarquis) [inline core/designsystem/src/test/kotlin/com/google/samples/apps/nowinandroid/core/designsystem/ThemeTest.kt:48]: Waiting on #1047 to remove this line
```suggestion
```

File: core/designsystem/src/test/kotlin/com/google/samples/apps/nowinandroid/core/designsystem/ThemeTest.kt (hunk 1 above)
Comment (dturner) [reply core/designsystem/src/test/kotlin/com/google/samples/apps/nowinandroid/core/designsystem/ThemeTest.kt:60]: #1047 has been merged

File: core/designsystem/src/test/kotlin/com/google/samples/apps/nowinandroid/core/designsystem/ThemeTest.kt (hunk 2 above)
Comment (dturner) [reply core/designsystem/src/test/kotlin/com/google/samples/apps/nowinandroid/core/designsystem/ThemeTest.kt:48]: #1047 has been merged

**Comment by SimonMarquis** [comment]
@dturner this PR is ready to be reviewed 🙇

**Review by dturner — APPROVED** [review]

**Pull request merged**